protoc-gen-swagger/genswagger/atlas_patch.go

```diff
@@ -508,4 +516,160 @@ func gatherRefs(v interface{}) map[string]struct{} {
 
 func refToName(ref string) string {
 	return strings.TrimPrefix(ref, "#/definitions/")
-}
\ No newline at end of file
+}
+
+func applyCustomAnnotations(defs spec.Definitions) spec.Definitions {
+	for k, v := range defs {
+		v.Description = v.Title + v.Description
+		v.Title = ""
+
+		if title, ok := getAnnotationValue(v.Description, titleAnnotation); ok {
+			v.Title, ok = title.(string)
+			if !ok {
+				log.Printf("unsupported title type in %s\n", v.Type)
+			}
+
+			v.Description = removeSpecials(v.Description, titleAnnotation)
+
+			defs[k] = v
+		}
+
+		if example, ok := getAnnotationValue(v.Description, exampleAnnotation); ok {
+			v.Example = example
+			v.Description = removeSpecials(v.Description, exampleAnnotation)
+
+			defs[k] = v
+		}
+
+		for fk, fv := range v.Properties {
+			fv.Description = fv.Title + fv.Description
+			fv.Title = ""
+
+			if example, ok := getAnnotationValue(fv.Description, exampleAnnotation); ok {
+				fv.Example = example
+				fv.Description = removeSpecials(fv.Description, exampleAnnotation)
+
+				defs[k].Properties[fk] = fv
+			}
+		}
+
+		if v.Definitions != nil {
+			v.Definitions = applyCustomAnnotations(v.Definitions)
+		}
+	}
+
+	return defs
+}
+
+func removeSpecials(comment, annotation string) (res string) {
+	lines := strings.Split(comment, "\n")
+	for i := 0; i < len(lines); i++ {
+		trimmed := strings.TrimSpace(lines[i])
+		if strings.HasPrefix(trimmed, annotation) && strings.HasSuffix(trimmed, "<<<EOF") {
+			pos := stringPosition(lines, "EOF", i)
+			if pos != -1 && pos < len(lines) {
+				lines = append(lines[:i], lines[pos+1:]...)
+			}
+			lines = append(lines[:i])
+
+		} else if strings.HasPrefix(trimmed, annotation) {
+			lines = append(lines[:i], lines[i+1:]...)
+		}
+	}
+
+	return strings.Join(lines, "\n")
+}
+
+func getAnnotationValue(comment, prefix string) (interface{}, bool) {
+	var (
+		value string
+		lines = strings.Split(comment, "\n")
+	)
+
+	for i := 0; i < len(lines); i++ {
+		line := strings.TrimSpace(lines[i])
+
+		if strings.HasPrefix(line, prefix) && strings.HasSuffix(line, "<<<EOF") {
+			pos := stringPosition(lines, "EOF", i)
+			if pos != -1 && pos < len(lines) {
+				value = strings.Join(lines[i+1:pos], "\n")
+			} else {
+				value = strings.Join(lines[i+1:], "\n")
+			}
+
+		} else if strings.HasPrefix(line, prefix) {
+			value = strings.TrimSpace(strings.TrimPrefix(line, prefix))
+		}
+
+		if value != "" {
+			break
+		}
+	}
+
+	if value != "" {
```

Comment:
Did you tried to put it just as raw json (byte array)? If for some reason you want to have total control over data in annotations, I would recommend to require quoting for strings (because for now "2" and 2 are indistinguishable)